src/Compiler.php

```diff
@@ -866,17 +1008,22 @@ private static function scanObject($input, &$object)
 
     private static function isPathToken($token)
     {
-        return is_array($token) && ($token[0] === Parser::TYPE_PATH);
+        return is_array($token) && count($token) > 0 && ($token[0] === Parser::TYPE_PATH);
```

Comment:
We should use parentheses here:
```php
... && (count($token) > 0) && ...
```

Otherwise, we're depending on the details of the operator precedence rules in PHP (which aren't exactly the same as in other languages).

This code works fine, but it _would_ reduce the cognitive burden on the reader if the evaluation were made explicit with parentheses.

Also! I've been called in on a number of occasions to help troubleshoot code where an expression that _looked_ fine was actually executed in an unexpected way... which lead to a very hard-to-find bug in production. You can make those bugs impossible simply by using an explicit order of evaluation.